0029-stream/0029-stream.md

```diff
@@ -436,6 +444,24 @@ In other words, if a sender resends data (e.g. because a packet was lost), it MU
 
 Asset details exposed by this frame MUST NOT change during the lifetime of a Connection.
 
+#### 5.3.15. `StreamReceipt` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Stream ID | VarUInt | Identifier of the stream this frame refers to. |
+| Receipt | 65-Byte OctetString | Proof provided by the receiver of the total amount received on this stream |
+
+The `Receipt` MUST contain the following fields encoded using the [Octet Encoding Rules (OER)](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview_of_OER.pdf):
+
+| Field | Type | Description |
+|---|---|---|
+| HMAC | UInt256 | HMAC-SHA256 over all other fields using the 32 byte Receipt Secret, which is pre-shared between the verifying party and the receiver. |
```

Comment:
Nit: if the `HMAC` was the last field instead of the first, it would be easy to implement serialization with one fewer allocation.

Reply:
I think I put the hmac first with verification in mind, but it makes sense to me to optimize the serialization.

Reply:
addressed in https://github.com/interledger/rfcs/pull/570/commits/8df93049769a21a0a8cd59b3d43902574b88e4ce